src/utilities/find_normfactors_from_cylinder_data.cxx

```diff
@@ -0,0 +1,158 @@
+      cylinder_projdata_ptr->get_proj_data_info_sptr()->get_LOR(lor, bin);
+      LORAs2Points<float> lor_as2points(lor);
+      LORAs2Points<float> intersection_coords;
+      if (find_LOR_intersections_with_cylinder(intersection_coords, lor_as2points, R) ==Succeeded::yes)
+      { //this only succeeds if LOR is intersecting with the cylinder
+        float N_lor = cylinder_viewgram[ax][tang]; //counts seen by this lor
+        c1 = intersection_coords.p1();
+        c2 = intersection_coords.p2();
+        float c12 = sqrt( pow(c1.z()-c2.z(), 2)    // length of intersection of lor with the cylinder
+                        + pow(c1.y()-c2.y(), 2)
+                        + pow(c1.x()-c2.x(), 2) );
+        if (c12>0.5) // if LOR intersection is lager than 0.5 mm, check the count per LOR
+        {
+          float N_lor_corrected=N_lor/c12; // corrected for the length
+          total_count+=N_lor_corrected;
+          num_active_LORs+=1;
+          if (N_lor_corrected<min_count && N_lor_corrected!=0) min_count=N_lor_corrected;
+        }
+      }
+    }
+  }
+  average_count=total_count/num_active_LORs;
+  std::cout<<"num_lor, tot_count_per_length_unit, average_count_per_length_unit, non_zero_min_per_length_unit = "<<num_active_LORs<<", "<<total_count<<", "<<average_count<<", "<<min_count<<"\n";
+
+  // find the norm factor per LOR
+  for (int seg =cylinder_projdata_ptr->get_min_segment_num(); seg <=cylinder_projdata_ptr->get_max_segment_num(); ++seg)
+    for (int view =cylinder_projdata_ptr->get_min_view_num(); view <=cylinder_projdata_ptr->get_max_view_num(); ++view)
+  {
+    Viewgram<float> cylinder_viewgram = cylinder_projdata_ptr->get_viewgram(view, seg);
+    Viewgram<float> out_viewgram = cylinder_projdata_ptr->get_empty_viewgram(view, seg);
+    for (int ax =cylinder_projdata_ptr->get_min_axial_pos_num(seg); ax <=cylinder_projdata_ptr->get_max_axial_pos_num(seg); ++ax)
+      for (int tang =cylinder_projdata_ptr->get_min_tangential_pos_num(); tang <=cylinder_projdata_ptr->get_max_tangential_pos_num(); ++tang)
+    {
+      Bin bin(seg, view, ax, tang);
+      cylinder_projdata_ptr->get_proj_data_info_sptr()->get_LOR(lor, bin);
+      LORAs2Points<float> lor_as2points(lor);
+      LORAs2Points<float> intersection_coords;
+      float NF_lor;
+      if (find_LOR_intersections_with_cylinder(intersection_coords, lor_as2points, R) ==Succeeded::yes)
+      { //this only succeeds if LOR is intersecting with the cylinder
+
+        /*
+          for each lor
+            find_LOR_intersections_with_cylinder => c1 & c2
+            c12 = |c1-c2| = sqrt(Δx^2+Δy^2+Δz^2)
+            N_lor/c12 should be the same for all therefore:
+            NF_lor= <N> / (N_lor/c12)
+        */
+
+        float N_lor = cylinder_viewgram[ax][tang]; //counts seen by this lor
+        c1 = intersection_coords.p1();
+        c2 = intersection_coords.p2();
+        float c12 = sqrt( pow(c1.z()-c2.z(), 2)    // length of intersection of lor with the cylinder
```

Comment:
it seems that this calculation does not take attenuation into account?

Reply:
No, it does not. The input data should be pre-corrected for attenuation, scatter and random.